### ISSUE TITLE
Make particles sidebar scroll separately from layout manager

### ISSUE THREAD

**Issue by alisamii** [gantry/gantry5#347]
If the layout you are working on is long, even on a large 27" screen, you can end up with the particles out of view. This makes the drag-and-drop less efficient as you have to scroll up, find the particle you want to add, then drag the particle and cause the layout manager to scroll by drag. This is visible in the screenshot below, taken on a screen with a viewport resolution of 1262x1339.

![particles-menu-scrolling](https://cloud.githubusercontent.com/assets/1498512/7788133/b4567efa-022e-11e5-8226-e16106b9dc2b.png)

It would be good if the layout manager and the particles scrolled independently of each other, thus you could scroll the layout to the operation you want, and then scroll (if needed) the particles to make the particle you want to add visible and then drag and drop.

**Comment by mahagr** [comment]
I would so much love to see this one. Scrolling down with mouse wheel works, but it is really slow.

**Comment by alisamii** [comment]
On a trackpad, you don't have mouse wheels, and click-dragging is a real drag, pun intended.

**Comment by alisamii** [comment]
@w00fz @kevlardev @mahagr

I think that this is a major usability enhancement that should be integrated sooner rather than later.

Layouts can easily get quite long with different sections and rows per section. As @mahagr said, you can scroll with the mouse wheel, but it is slow and inconsistent. On a trackpad, it is even more complicated.

@w00fz explained that due to the drag-and-drop nature of this page, a dual-scroll layout may be complicated and present other issues.

Another option would be to make the particles pane be sticky, with collapsible groups of particles (some logic may be needed to help group them appropriately) and possible even making the particle "drag-and-droppable" pill smaller to conserve screen real estate. Another option may be to widen the left particles pane and display the particles in 2 columns.

The long and short of it (pun intended) is that a long layout manager on a short screen (think 13" screen) makes it quite challenging to drag and drop the particle in the correct row without scrolling and/or finger acrobatics (on a trackpad).

I hope that this can be included in RC3, and strongly urge that it be considered.

**Comment by federone** [comment]
hi, 

i was playing with this, and in "administrator/components/com_gantry5/js/main.js" added this:

```
(function($){$(window).load(function(){
	// add css
		$('head').append('<style type="text/css">.fixed_picker{overflow-y:scroll;height:95%;position:fixed;top:5%;left:0;width:14%;background:#EEE;padding:1%}</style>');
	// scroll
	$(document).scroll(function() {    
		var scroll = $(this).scrollTop();
		if (scroll >= 300) {
			$(".g5-lm-particles-picker").addClass('fixed_picker');
		} else {
			$(".g5-lm-particles-picker").removeClass('fixed_picker');
		}
	});
})})(jQuery);
```

and works well ... what do you think?

**Comment by mahagr** [comment]
This one will not make into RC3 as it takes some time to fix and we have had (and we still have) more severe issues which need our attention.

That said, I totally agree with you and wish that this one will be considered soon.

**Comment by sural98** [comment]
And we also need floating buttons for Save Layout, Clear, Load, Save Styles, Recompile CSS

**Comment by gyvermac** [comment]
:+1:    I really think it'll be a serious improvement to be able to easily drag and drop without scrolling.
Dual-scroll layout is certainly the best way (for the user !) but any other solution like collapsible group of particules can make the job.

**Comment by pdelbar** [comment]
+1

**Comment by w00fz** [comment]
Ok this is now implemented. Was even harder to implement than I originally anticipated but bottom line is, you now have a fixed particle picker that follows you as you scroll.

It simulates the `sticky` CSS3 property so while at the top or if you reach the bottom it will end being fixed.

I also did a minor UX/UI revamp. The sidebar is now slightly smaller, the particles in the picker are also a bit smaller (padding / font size) and spaced out a bit better.

Enjoy!
